src/components/ListView.js

```diff
@@ -7,12 +7,48 @@ const ListView = ({ shoppingList, loading, error }) => {
   const [shoppingListEmpty, setShoppingListEmpty] = useState(true);
   const [length, setLength] = useState(0);
   const [value, setValue] = useState('');
+  const [sortedList, setSortedList] = useState([]);
+  //indicator for urgency status of item
+  let purchaseIndex;
 
   const handleChange = (e) => setValue(e.target.value);
 
+  //assign value to purchaseIndex based on these parameters
+  const checkIndex = (lastPurchase, nextPurchase, timesPurchased) => {
+    //check interval between current date and date of last purchase
+    //set index as inactive if item has been purchased only once or less
+    //or if interval is >= 2 times the projected estimate
+    const interval = new Date().getTime() - lastPurchase;
+    if (timesPurchased <= 1 || interval >= 2 * nextPurchase * 86400000) {
+      purchaseIndex = 'inactive';
+    } else if (nextPurchase < 7) {
+      purchaseIndex = 'soon';
+    } else if (nextPurchase >= 7 && nextPurchase < 30) {
+      purchaseIndex = 'kind-of-soon';
+    } else if (nextPurchase > 30) {
+      purchaseIndex = 'not-soon';
+    }
+  };
+
+  //todays date - lastpurchase >= 2*nextpurchase
+
   useEffect(() => {
+    //sort items according to days left till next purchase
+    //if items have same number of days left, filter alphabetically
+    //used localeCompare to sort without consideration for case
+    const filterUrgency = () => {
```

Comment:
if the items get sorted, then, probably, it might be a good idea to rename this function to sortByUrgency for better readability?

Reply:
Okay